src/lib.rs

```diff
@@ -1061,17 +1065,177 @@ fn digraph_astar_shortest_path(
     Ok(out_path.to_object(py))
 }
 
-/// The provided node is invalid.
+/// Return a :math:`G_{np}` directed random graph, also known as an
+/// Erdős-Rényi graph or a binomial graph.
+///
+/// The :math:`G_{n,p}` graph algorithm chooses each of the
+/// :math:`n (n - 1)` possible edges with probability :math:`p`.
+/// This algorithm [1]_ runs in :math:`O(n + m)` time, where :math:`m` is the
+/// expected number of edges, which equals :math:`p n (n - 1)/2`.
+///
+/// Based on the implementation of the networkx function
+/// ``fast_gnp_random_graph`` [2]_
+///
+/// :param int num_nodes: The number of nodes to create in the graph
+/// :param float probability: The probability of an edge
+/// :param int seed: An optional seed to use for the random number generator
+///
+/// :return: A PyDiGraph object
+/// :rtype: PyDiGraph
+///
+/// .. [1] Vladimir Batagelj and Ulrik Brandes,
+///    "Efficient generation of large random networks",
+///    Phys. Rev. E, 71, 036113, 2005.
+/// .. [2] https://github.com/networkx/networkx/blob/networkx-2.4/networkx/generators/random_graphs.py#L49-L120
+#[pyfunction]
+#[text_signature = "(num_nodes, probability, seed=None, /)"]
+pub fn directed_gnp_random_graph(
+    py: Python,
+    num_nodes: isize,
+    probability: f64,
+    seed: Option<u64>,
+) -> PyResult<digraph::PyDiGraph> {
+    if num_nodes <= 0 {
+        return Err(IndexError::py_err("num_nodes must be > 0"));
```

Comment:
This is a minor thing, but wouldn't a ValueError make more sense since `num_nodes` is not really used as index?

Reply:
Oh, yeah that's probably a better error to return I'll change it. I was struggling to think of what a good exception would be for this. `ValueError`  is better choice

Reply:
Done in: https://github.com/Qiskit/retworkx/pull/94/commits/05464c6bb6207dc0e243c63d22c9ae698e95a099